common/io_lock.c

```diff
@@ -0,0 +1,75 @@
+#include "io_lock.h"
+#include <assert.h>
+#include <ccan/io/io_plan.h>
+
+struct io_lock {
+	bool locked;
+};
+
+/* Struct to hold information while we wait for the lock to be freed */
+struct io_lock_waiter {
+	struct io_plan *(*next)(struct io_conn *conn, void *next_arg);
```

Comment:
Right, we haven't done this so far, but it might increase legibility. @rustyrussell would that be Rusty-Style C? :wink:

Reply:
No, I don't typedef structs, unless I really don't want you to treat them differently (eg. they're structs on some platforms, and ints in others, or int on some, long on others, etc.).

When I see 'struct foo' I know:
1. I can assign it, but not compare it naively.
2. I should pass it by pointer, rather than copy.
3. It can be predeclared with 'struct foo' to avoid pulling in the entire definition.